### PR TITLE
add PMD and start cleaning up its findings, #623

### PR DESCRIPTION
**Pull request by ThrawnCA**
Technically I have changed code, but since it was just things like removing unused imports etc, there should be nothing that would need to go in a changelog (if any changes are externally visible, then I made a mistake).
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [ ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Comment by ThrawnCA** [comment]
:D Well, it looks like my cleanup wasn't clean *enough*.

Incidentally, I notice that `SourceDirectoryWizard` claims to be generated in some fashion, and that the fields shouldn't be altered...is that actually part of the build, or was it generated once and now maintained manually?

File: spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java [hunk 1]
```diff
@@ -19,11 +19,7 @@
 
 package edu.umd.cs.findbugs.ba.generic;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (KengoTODA) [inline spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java]: I think we do not need to use wildcard to import, at least it's not enabled in [Eclipse setting](https://github.com/spotbugs/spotbugs/blob/release-3.1/gradle/.settings-templates/org.eclipse.jdt.ui.prefs).

Comment (ThrawnCA) [reply spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java]: Well, I can switch off that PMD rule, but it means writing a custom PMD rules file...

How would you feel about me introducing AssertJ? Then only one static import is needed.

Comment (KengoTODA) [reply spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java]: I personally don't care which tool we use.

Comment (iloveeclipse) [reply spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java]: I would avoid adding extra dependencies just to fix some cosmetic issue.

Comment (ThrawnCA) [reply spotbugs/src/test/java/edu/umd/cs/findbugs/ba/generic/TestGenericObjectType.java]: @iloveeclipse Fair, although AssertJ is actually quite a nice testing tool in its own right.

But I've written a custom rules file and disabled the static import limitation, anyway.

**Pull request merged**